data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetAttestationData.java

```diff
@@ -0,0 +1,118 @@
+import static tech.pegasys.teku.beaconrestapi.SingleQueryParameterUtils.getParameterValueAsUInt64;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.javalin.http.Context;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.ValidatorDataProvider;
+import tech.pegasys.teku.api.response.v1.validator.GetAttestationDataResponse;
+import tech.pegasys.teku.api.schema.Attestation;
+import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class GetAttestationData extends AbstractHandler {
+  public static final String ROUTE = "/eth/v1/validator/attestation_data";
+
+  private final ValidatorDataProvider provider;
+
+  public GetAttestationData(final DataProvider provider, final JsonProvider jsonProvider) {
+    this(provider.getValidatorDataProvider(), jsonProvider);
+  }
+
+  public GetAttestationData(final ValidatorDataProvider provider, final JsonProvider jsonProvider) {
+    super(jsonProvider);
+    this.provider = provider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Requests that the beacon node produce an AttestationData.",
+      tags = {TAG_VALIDATOR},
+      queryParams = {
+        @OpenApiParam(
+            name = SLOT,
+            description = "`uint64` The slot for which an attestation data should be created.",
+            required = true),
+        @OpenApiParam(
+            name = COMMITTEE_INDEX,
+            type = Integer.class,
+            description =
+                "`Integer` The committee index for which an attestation data should be created.",
+            required = true)
+      },
+      description =
+          "Returns attestation data for the block at the specified non-finalized slot.\n\n"
+              + "This endpoint is not protected against slashing. Signing the returned attestation data can result in a slashable offence.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetAttestationDataResponse.class)),
+        @OpenApiResponse(status = RES_BAD_REQUEST, description = "Invalid parameter supplied"),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR),
+        @OpenApiResponse(status = RES_SERVICE_UNAVAILABLE, description = SERVICE_UNAVAILABLE)
+      })
+  @Override
+  public void handle(Context ctx) throws Exception {
+    try {
```

Comment:
need to check syncing up front and return the 503 if its syncing...
something like in PostBlock:
```
      if (syncDataProvider.getSyncStatus().is_syncing) {
        ctx.status(SC_SERVICE_UNAVAILABLE);
        ctx.result(BadRequest.serviceUnavailable(jsonProvider));
        return;
      }
```

Reply:
That's done in `ValidatorApiHandler` which throws a `NodeSyncingException` which we've got a default exception handler setup for.

Reply:
ah ok - cool then 👍